src/include/platform/OTARequestorInterface.h

```diff
@@ -133,10 +133,48 @@ class ProviderLocationList
         return CHIP_ERROR_NOT_FOUND;
     }
 
+    /**
+     * Load the DefaultOTAProvidersList attribute value from the platform KVS
+     * TODO: Use DefaultStorageKeyAllocator.AttributeValue() to get a key composed of the concrete attribute path
+     */
+    CHIP_ERROR LoadFromPersistentStorage(OTARequestorStorage * storage)
+    {
+        size_t read_bytes_size = 0;
+        uint16_t max_size =
+            static_cast<uint16_t>(mMaxSize * sizeof(app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type));
+        CHIP_ERROR ret = storage->SyncGetKeyValue(defaultOtaProvidersListkey, mList, max_size, &read_bytes_size, 0);
+        if (ret == CHIP_NO_ERROR)
+        {
+            mListSize = read_bytes_size / sizeof(app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type);
+        }
+        return ret;
+    }
+
+    /**
+     * Store the DefaultOTAProvidersList attribute value from the platform KVS
+     * TODO: Use DefaultStorageKeyAllocator.AttributeValue() to get a key composed of the concrete attribute path
+     */
+    CHIP_ERROR StoreIntoPersistentStorage(OTARequestorStorage * storage)
+    {
+        return storage->SyncSetKeyValue(
+            defaultOtaProvidersListkey, mList,
+            static_cast<uint16_t>(mListSize * sizeof(app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type)));
+    }
+
+    /**
+     * Delete the DefaultOTAProvidersList attribute from the platform KVS
+     * TODO: Use DefaultStorageKeyAllocator.AttributeValue() to get a key composed of the concrete attribute path
+     */
+    CHIP_ERROR DeleteFromPersistentStorage(OTARequestorStorage * storage)
+    {
+        return storage->SyncDeleteKeyValue(defaultOtaProvidersListkey);
+    }
+
 private:
-    Optional<app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type> mList[CHIP_CONFIG_MAX_FABRICS];
-    size_t mListSize = 0;
-    size_t mMaxSize  = CHIP_CONFIG_MAX_FABRICS;
+    app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type mList[CHIP_CONFIG_MAX_FABRICS] = {};
```

Comment:
I removed the "Optional" nature of this array to make it easier to copy buffers from/to the KVS, without having an intermediate buffer. @carol-apple